sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/MonotonicallyIncreasingID.scala

```diff
@@ -40,13 +40,14 @@ import org.apache.spark.sql.types.{DataType, LongType}
       represent the record number within each partition. The assumption is that the data frame has
       less than 1 billion partitions, and each partition has less than 8 billion records.""",
   extended = "> SELECT _FUNC_();\n 0")
-case class MonotonicallyIncreasingID() extends LeafExpression with Nondeterministic {
+case class MonotonicallyIncreasingID(offset: Long = 0) extends LeafExpression
```

Comment:
Check if this still works in SQL. We might have to change offset into a literal expression. See HyperLogLogPlusPlus or NTile for examples of this.